### ISSUE TITLE
skylight: lib/mri/mkmf.rb:1299:in `find_type': wrong number of arguments (given 1, expected 2+) (ArgumentError)

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1896]
To reproduce:

```
gem install skylight
```

Error:

```Building native extensions. This could take a while...
ERROR:  Error installing skylight:
	ERROR: Failed to build gem native extension.

    current directory: ~/.gem/truffleruby/2.6.5/gems/skylight-4.2.1/ext
~/.rubies/truffleruby-20.1.0-dev/bin/truffleruby -I ~/.rubies/truffleruby-20.1.0-dev/lib/mri -r ./siteconf20200208-39800-1h6f77a.rb extconf.rb
~/.rubies/truffleruby-20.1.0-dev/lib/mri/mkmf.rb:1299:in `find_type': wrong number of arguments (given 1, expected 2+) (ArgumentError)
	from ~/.rubies/truffleruby-20.1.0-dev/lib/truffle/ffi/struct.rb:272:in `find_type'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/truffle/ffi/struct.rb:266:in `find_field_type'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/truffle/ffi/struct.rb:306:in `array_layout'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/truffle/ffi/struct.rb:218:in `layout'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/truffle/socket/truffle.rb:63:in `config'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/truffle/socket/truffle/foreign/addrinfo.rb:31:in `Addrinfo'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/truffle/socket/truffle/foreign/addrinfo.rb:30:in `Foreign'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/truffle/socket/truffle/foreign/addrinfo.rb:29:in `Socket'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/truffle/socket/truffle/foreign/addrinfo.rb:28:in `Truffle'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/truffle/socket/truffle/foreign/addrinfo.rb:27:in `<top (required)>'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/mri/rubygems/core_ext/kernel_require.rb:54:in `require'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/mri/rubygems/core_ext/kernel_require.rb:54:in `require'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/truffle/socket.rb:38:in `<top (required)>'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/mri/rubygems/core_ext/kernel_require.rb:54:in `require'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/mri/rubygems/core_ext/kernel_require.rb:54:in `require'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/mri/net/protocol.rb:22:in `<top (required)>'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/mri/net/http.rb:23:in `require_relative'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/mri/net/http.rb:23:in `<top (required)>'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/mri/rubygems/core_ext/kernel_require.rb:54:in `require'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/mri/rubygems/core_ext/kernel_require.rb:54:in `require'
	from ~/.gem/truffleruby/2.6.5/gems/skylight-4.2.1/lib/skylight/native_ext_fetcher.rb:3:in `<top (required)>'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/mri/rubygems/core_ext/kernel_require.rb:54:in `require'
	from ~/.rubies/truffleruby-20.1.0-dev/lib/mri/rubygems/core_ext/kernel_require.rb:54:in `require'
	from extconf.rb:9:in `<main>'
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=~/.rubies/truffleruby-20.1.0-dev/bin/truffleruby
log file does not exist

extconf failed, exit code 1

Gem files will remain installed in ~/.gem/truffleruby/2.6.5/gems/skylight-4.2.1 for inspection.
Results logged to ~/.gem/truffleruby/2.6.5/extensions/x86_64-darwin/20.1.0-dev-9e0b0468/skylight-4.2.1/gem_make.out

```

**Comment by eregon** [comment]
Interesting case where code in FFI ends up conflicting with code in mkmf.rb.
I'll need to find out why it doesn't happen on MRI.

**Comment by eregon** [comment]
```
ruby -rmkmf -rffi -e 'module A; class A < FFI::Struct; layout :a, :int; end; end'
```
reproduces and also fails in MRI.

So basically it's an issue of FFI: https://github.com/ffi/ffi/blob/33fb45a80e20fd9d9c39de3753abca9b4ef3abe7/lib/ffi/struct.rb#L247
Because mkmf defines a public `find_type` on every object by `include MakeMakefile`.

The only reason it doesn't fail on MRI for the `extconf.rb` is that MRI doesn't use FFI internally for the `socket` stdlib.

The require chain here is `extconf.rb -> net/http -> socket -> ffi` and in #2010 it's `extconf.rb -> openssl -> ipaddr -> socket -> ffi`.

**Comment by eregon** [comment]
Fixed by a54e85bbd68e3929781fc1654a2c1ce71e782a37.
`gem install skylight` works for me locally now.
